lbr/lbr.py

```diff
@@ -235,6 +235,31 @@ def mrn_cond(l):
     elif v2ii2v_dst: inc_stat('V2I transition-Penalty')
   if xinfo.is_cond_br() and xinfo.is_taken():
     LC.glob['cond_%sward-taken' % ('for' if ip > xip else 'back')] += 1
+  #Shortcircuit detection
+  if 'set' in x86.get('inst',line):
+    x = len(lines)-1
+    max_inst=8
+    max_cnt=0
+    candidate_regs=[]
+    entry_pt=0
+    while x > 0 and not x86.is_branch(lines[x],x86.CALL_RET) and  max_cnt <= max_inst:
+      if re.search(x86.TEST_CMP,lines[x]):
+        if entry_pt == 0:
+          candidate_regs=x86.get("srcs",lines[x]) + [x86.get("dst",lines[x])]
+          for i in candidate_regs:
+            if '0x' in i:
```

Comment:
So its filtering out random immediate values as possible candidate registers. For example it will consume "cmp rax,0x1000" and will have [rax,0x1000] in the list, so that removes the 0x1000

Reply:
Oh.. then this is not generic enough; e.g. what if the assembler did not put the immediate value in hexa?